web/src/components/AddPluginConfigurationPage.js

```
@@ -0,0 +1,216 @@
+import React from 'react';
+import AppBar from 'material-ui/AppBar';
+import PropTypes from 'prop-types';
+import {
+    CircularProgress, DatePicker, Dialog, FlatButton, IconButton, MenuItem, RaisedButton, SelectField, Snackbar,
+    TextField
+} from 'material-ui';
+import {NavigationArrowBack} from 'material-ui/svg-icons';
+
+const style = {
+    body: {
+        marginLeft: 16
+    }
+};
+
+class AddPluginConfigurationPage extends React.Component {
+
+    constructor(props, context) {
+        super(props, context);
+        this.state = {
+            configurationName: '',
+            enteredConfiguration: {},
+            showConfigurationAdded: false,
+            invalidInputDialogOpen: false
+        };
+    }
+
+    render() {
+        return this._pluginsLoaded() ? this._showData() : this._showLoading();
+    }
+
+    _pluginsLoaded() {
+        return !this.props.loading && Object.keys(this.props.plugin).length !== 0;
```

this means that if you don't find a plugin in https://github.com/novoda/dashboards/pull/71/files#diff-ba817a5d2f45f4cac90db9f46fe8d309R10 you'll always have a loading state, maybe fail fast in the mapping?